resources/regexes.ts

```diff
@@ -18,32 +20,86 @@ const defaultParams = <
   T extends LogDefinitionTypes,
   V extends LogDefinitionVersions,
 >(type: T, version: V, include?: string[]): Partial<ParseHelperFields<T>> => {
-  include ??= Object.keys(logDefinitionsVersions[version][type].fields);
-  const params: { [index: number]: { field: string; value?: string; optional: boolean } } = {};
-  const firstOptionalField = logDefinitionsVersions[version][type].firstOptionalField;
+  const logType = logDefinitionsVersions[version][type];
+  if (include === undefined) {
+    include = Object.keys(logType.fields);
+    if ('repeatingFields' in logType) {
+      include.push(logType.repeatingFields.label);
+    }
+  }
+
+  const params: {
+    [index: number]: {
+      field: string;
+      value?: string;
+      optional: boolean;
+      repeating?: boolean;
+      repeatingKeys?: string[];
+      sortFn?: (L: unknown, R: unknown) => number;
+    };
+  } = {};
+  const firstOptionalField = logType.firstOptionalField;
 
-  for (const [prop, index] of Object.entries(logDefinitionsVersions[version][type].fields)) {
+  for (const [prop, index] of Object.entries(logType.fields)) {
     if (!include.includes(prop))
       continue;
-    const param: { field: string; value?: string; optional: boolean } = {
+    const param: { field: string; value?: string; optional: boolean; repeating?: boolean } = {
       field: prop,
       optional: firstOptionalField !== undefined && index >= firstOptionalField,
     };
     if (prop === 'type')
-      param.value = logDefinitionsVersions[version][type].type;
+      param.value = logType.type;
 
     params[index] = param;
   }
 
+  if ('repeatingFields' in logType && include.includes(logType.repeatingFields.label)) {
+    params[logType.repeatingFields.startingIndex] = {
+      field: logType.repeatingFields.label,
+      optional: firstOptionalField !== undefined &&
+        logType.repeatingFields.startingIndex >= firstOptionalField,
+      repeating: true,
+      repeatingKeys: [...logType.repeatingFields.names],
+      sortFn: logType.repeatingFields.sortFn,
+    };
+  }
+
   return params as unknown as Partial<ParseHelperFields<T>>;
 };
+type RepeatingFieldsMap<
+  TBase extends LogDefinitionTypes,
+  TKey extends RepeatingFieldsTypes = TBase extends RepeatingFieldsTypes ? TBase : never,
+> = {
+  // This syntax is a bit weird to make typescript compiler happy
+  [
+    label in RepeatingFieldsDefintions[TKey]['repeatingFields'][
+      'label'
+    ] as `${RepeatingFieldsDefintions[TKey]['repeatingFields']['label']}`
```

Comment:
Backticks are a string evaluation syntax just like normal JS/TS strings.

Basically, as far as I'm aware, you can't read the value of a type even when it's const, and extend that into defining the key/property on another type.

Consider assigning the value of a dynamic key to an object in JS:
```js
const someObjectDefinition = { 'key': 'pair' };
const value = 'anything';
const obj = {};
obj[someObjectDefinition.key] = value;
```

In typescript, this last operation isn't possible when defining a type directly (as opposed to assigning to an object). Instead, the code in `regexes.ts`/`netregexes.ts` does this equivalent logic:
```js
const someObjectDefinition = { 'key': 'pair' };
const value = 'anything';
const obj = {};
for (const keyDef in [someObjectDefinition.key]) {
  obj[keyDef] = value;
}
```

In other words, this is read as a string union with only one entry:
```ts
RepeatingFieldsDefintions[TKey]['repeatingFields']['label']
```
This is a loop over that one-element union:
```ts
label in ...
```
And this forces the value to be read as a constant entry so that the actual typedef gets a proper key (instead of an indexer):
```ts
 as `${RepeatingFieldsDefintions[TKey]['repeatingFields']['label']}`
```

---

It would be easier if this was allowed, but it's not valid TS syntax:
```ts
type RepeatingFieldsMap<
  TBase extends LogDefinitionTypes,
  TKey extends RepeatingFieldsTypes = TBase extends RepeatingFieldsTypes ? TBase : never,
> = {
  `${RepeatingFieldsDefintions[TKey]['repeatingFields']['label']}`: {
    [name in RepeatingFieldsDefintions[TKey]['repeatingFields']['names'][number]]:
      | string
      | string[];
  }[];
}
```

Reply:
> And this forces the value to be read as a constant entry so that the actual typedef gets a proper key (instead of an indexer):

It is this part that I don't understand.  It seems to me like all the type-checking is the same with or without that.  If I remove it, the regex still typechecks `pair` vs incorrect `pair2` parameters.  Sorry, I'm just not following this one `as` bit here.

Reply:
I tested this and it wasn't working before. I'm not sure now if that was because of desync (due to running cactbot under WSL which causes some weird filewatcher issues) or if it's working now because of other changes that were made between when I originally tested and now.

Either way, I'll make this adjustment with the next push.